drivers/bluetooth/hci/h4.c

```diff
@@ -502,13 +502,35 @@ int __weak bt_hci_transport_setup(const struct device *dev)
 	return 0;
 }
 
+#if defined(CONFIG_BT_HCI_INIT)
+static int h4_init(void)
+{
+	/* Extern bt_h4_vnd_init function.
+	 * This function executes vendor-specific initialization sequence to
+	 * initialize BT Controller before BT HCI is opened.
+	 * bt_h4_vnd_setup function must be implemented in vendor-specific HCI
+	 * extansion module if CONFIG_BT_HCI_INIT is enabled.
+	 */
+	extern int bt_h4_vnd_init(const struct device *dev);
+
+	return bt_h4_vnd_init(h4_dev);
+}
+#endif
+
 static int h4_open(void)
 {
 	int ret;
 	k_tid_t tid;
 
 	LOG_DBG("");
 
+#if defined(CONFIG_BT_HCI_INIT)
+	ret = h4_init();
+	if (ret < 0) {
+		return ret;
+	}
+#endif
```

Comment:
Couldn't you just implement your own `bt_hci_transport_setup()` instead of adding yet another vendor "setup" function?

Reply:
Yes. I updated the PR.